File: java/ScanHistoryDAO.java
package com.pikatech.database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

@Dao
public interface ScanHistoryDAO {

    @Insert
    void insert(ScanHistory... scanHistories);

    @Update
    void update(ScanHistory... scanHistories);

    @Delete
    void delete(ScanHistory... scanHistories);


    @Query("update ScanHistory set server_transfer=:Status,layoutInput=:LyoutInput where time=:time ")
    void updateSendtoServer(long time,String Status,String LyoutInput);

    @Query("update ScanHistory set excel_transfer=:Status where time=:time ")
    void updateSendtoExcel(long time,String Status);

    @Query("delete FROM ScanHistory where time<:time ")
    void deleteOldTime(long time);

    @Query("SELECT * FROM ScanHistory where type= :Type order by time desc ")
    List<ScanHistory> getScanHistoryByType(String Type);
}

File: lib/scan_history/scan_history_dao.dart
import 'package:floor/floor.dart';

import 'scan_history.dart';

@dao
abstract class ScanHistoryDAO {
  @Query(
      'UPDATE ScanHistory set server_transfer = :status,layoutInput = :lyoutInput WHERE time = :time')
  Future<ScanHistory> updateSendtoServer(
      String status, String lyoutInput, double time);

  @Query("UPDATE ScanHistory set excel_transfer = :status WHERE time = :time")
  Future<void> updateSendtoExcel(String status, double time);

  @Query("DELETE FROM ScanHistory where time < :time")
  Future<void> deleteOldTime(double time);

  @Query("SELECT * FROM ScanHistory WHERE type = :type order by time desc")
  Future<List<ScanHistory>> findScanHistoryByType(String type);

  @insert
  Future<void> insertScanHistory(ScanHistory scanHistories);

  @update
  Future<void> updateScanHistory(ScanHistory scanHistories);

  @delete
  Future<void> deleteScanHistory(ScanHistory scanHistories);
}
